Sources/MapboxDirections/Incident.swift

```diff
@@ -21,46 +20,215 @@ public struct Incident: Codable, Equatable {
         case geometryIndexStart = "geometry_index_start"
         case geometryIndexEnd = "geometry_index_end"
     }
-
-    var identifier: String
-    var type: String
-    var description: String
-    var creationTime: String
-    var startTime: String
-    var endTime: String
-    var impact: String?
-    var subtype: String?
-    var subtypeDescription: String?
-    var alertCodes: [Int]
-    var lanesBlocked: [Int]
-    var geometryIndexStart: Int
-    var geometryIndexEnd: Int
+    
+    /// Defines known types of incidents.
+    ///
+    /// Each incident may or may not have specific set of data, depending on it's `kind`
+    public enum Kind: String {
+        case Accident = "accident"
+        case Congestion = "congestion"
+        case Construction = "construction"
+        case DisabledVehicle = "disabled_vehicle"
+        case LaneRestriction = "lane_restriction"
+        case MassTransit = "mass_transit"
+        case Miscellaneous = "miscellaneous"
+        case OtherNews = "other_news"
+        case PlannedEvent = "planned_event"
+        case RoadClosure = "road_closure"
+        case RoadHazard = "road_hazard"
+        case Weather = "weather"
+    }
+    
+    /// Defines a lane affected by the `Incident`
+    ///
+    /// Each `Incident` may have arbitrary of affected lanes
+    public enum BlockedLane: String, Codable {
+        /// Left lane
+        case left = "LEFT"
+        /// Left and center lanes
+        case leftCenter = "LEFT CENTER"
```

Comment:
According to [this presentation](https://slideplayer.com/slide/9372700/) (slide 11), traffic information management (TIM) systems in the U.S. use “left center lane” to refer to the second lane from left on a four-lane highway:

![](https://user-images.githubusercontent.com/1231218/101374246-c8198c00-3862-11eb-87bd-ee3f56cf794e.png)

Note that this slide isn’t globally applicable: for example, inside and outside would be reversed in everyday speech in Taiwan. But I don’t know if that would affect the incident data we get.